### PR TITLE
Incredibly old witch fix & Donkey Mount

### PR DESCRIPTION
**Pull request by heberholanda**
- Incredibly old witch - Fix time and effect from
- Fix Donkey Mount - Now check target Looktype equal MountType and valid target name
- Incredibly old witch - fix effect
- Incredibly old witch - change target to false

Incre 2

### CODE REVIEW

File: data/scripts/monsters/bosses/incredibly_old_witch.lua [hunk 1]
```diff
@@ -70,15 +70,15 @@ monster.loot = {
...
-	{name ="outfit", interval = 4000, chance = 12, range = 7, radius = 4, effect = CONST_ME_MAGIC_BLUE, target = true, duration = 2000, outfitMonster = "bug"},
-	{name ="outfit", interval = 4000, chance = 12, range = 7, radius = 4, effect = CONST_ME_MAGIC_BLUE, target = true, duration = 2000, outfitMonster = "pig"},
-	{name ="outfit", interval = 4000, chance = 12, range = 7, radius = 4, effect = CONST_ME_MAGIC_BLUE, target = true, duration = 2000, outfitMonster = "kongra"},
-	{name ="outfit", interval = 4000, chance = 12, range = 7, radius = 4, effect = CONST_ME_MAGIC_BLUE, target = true, duration = 2000, outfitMonster = "dog"},
-	{name ="outfit", interval = 4000, chance = 12, range = 7, radius = 4, effect = CONST_ME_MAGIC_BLUE, target = true, duration = 2000, outfitMonster = "skunk"},
-	{name ="outfit", interval = 4000, chance = 12, range = 7, radius = 4, effect = CONST_ME_MAGIC_BLUE, target = true, duration = 2000, outfitMonster = "donkey"}
+	{name ="outfit", interval = 4000, chance = 12, range = 7, radius = 4, effect = CONST_ME_TELEPORT, target = false, duration = 6000, outfitMonster = "rat"},
```
Comment (attack787) [inline data/scripts/monsters/bosses/incredibly_old_witch.lua]: Target is true.  This video show spell on target : https://youtu.be/R5D2CPcj-Ho?t=181
Duration of the spell is longer than 6 seconds, but I dont know how long.

File: data/scripts/actions/mounts/mounts.lua [hunk 2]
```diff
@@ -7,7 +7,7 @@ local config = {
...
 	[13305]	= {NAME = 'Crustacea Gigantica', ID = 7, BREAK = true, TYPE = TYPE_MONSTER, CHANCE = 40, FAIL_MSG = { {1, 'The crustacea has run away.'}, {2, 'The crustacea ate the shrimp.'} }, SUCCESS_MSG = 'You have tamed the crustacea.', ACHIEV = "Fried Shrimp"},
 	[13536] = {NAME = 'Crystal Wolf', ID = 16, BREAK = true, TYPE = TYPE_MONSTER, CHANCE = 40, FAIL_MSG = { {1, 'The wolf has run away.'} }, SUCCESS_MSG = 'You have tamed the wolf.', ACHIEV = "The Right Tone"},
-	[13537]	= {NAME = 'Donkey', ID = 13, BREAK = true, TYPE = TYPE_MONSTER, CHANCE = 40, FAIL_MSG = {	{1, 'The donkey transformation suddenly wears off.'}, {2,  'Heeee-haaa-haaa-haaw!'}, {3, 'You did not manage to feed the donkey enough apple slices.'} }, SUCCESS_MSG = 'Heeee-haaaaw!', ACHIEV = "Loyal Lad"},
+	[13537]	= {NAME = 'Donkey', ID = 13, BREAK = true, TYPE = TYPE_MONSTER, TARGET_NAME = "Incredibly Old Witch", TARGET_LOOKTYPE = 387, CHANCE = 40, FAIL_MSG = {	{1, 'The donkey transformation suddenly wears off.'}, {2,  'Heeee-haaa-haaa-haaw!'}, {3, 'You did not manage to feed the donkey enough apple slices.'} }, SUCCESS_MSG = 'Heeee-haaaaw!', ACHIEV = "Loyal Lad"},
```
Comment (debone) [inline data/scripts/actions/mounts/mounts.lua:10]: I think that only the looktype is required. In RL it's easier to lure a monster and let her transform it.

File: data/scripts/actions/mounts/mounts.lua [hunk 3]
```diff
@@ -82,14 +82,17 @@ function mounts.onUse(cid, item, fromPosition, itemEx, toPosition)
 			player:say('You can\'t tame a summon!', TALKTYPE_MONSTER_SAY)
 			return true
 		end
-
-		if mount.NAME == targetMonster:getName() then
+	
+		if mount.NAME == targetMonster:getName() or (mount.TARGET_NAME and mount.TARGET_NAME == targetMonster:getName()) then
+			if (mount.TARGET_LOOKTYPE and mount.TARGET_LOOKTYPE ~= targetMonster:getOutfit().lookType) then
```
Comment (debone) [inline data/scripts/actions/mounts/mounts.lua:87]: Maybe follow the pattern of the function and add a whole new section for specific mounts that need a outfit?

**Review by debone — CHANGES_REQUESTED** [review]

File: data/scripts/monsters/bosses/incredibly_old_witch.lua (hunk 1 above)
Comment (heberholanda) [reply data/scripts/monsters/bosses/incredibly_old_witch.lua]: Thx brow

File: data/scripts/actions/mounts/mounts.lua (hunk 2 above)
Comment (heberholanda) [reply data/scripts/actions/mounts/mounts.lua:10]: As I understand it, only in this "Incredibly Old Witch" can you deliver the mount.

File: data/scripts/actions/mounts/mounts.lua (hunk 3 above)
Comment (heberholanda) [reply data/scripts/actions/mounts/mounts.lua:87]: I haven't yet found another mount with the same scenario.

Comment (debone) [reply data/scripts/actions/mounts/mounts.lua:87]: Still, it took me a while to understand this logic as it's mixed with the monster one :(

File: data/scripts/actions/mounts/mounts.lua (hunk 2 above)
Comment (debone) [reply data/scripts/actions/mounts/mounts.lua:10]: Can you confirm with the videos? I'm quite sure that you can do with any monster that the witch transforms into a donkey.

Comment (heberholanda) [reply data/scripts/actions/mounts/mounts.lua:10]: Only target name =Incredibly Old Witch, Did you test it?

Comment (debone) [reply data/scripts/actions/mounts/mounts.lua:10]: No, I meant that on the global you don't need to take the mount from the witch. If she transforms a rat in a donkey, you can get the donkey from the rat.

Comment (heberholanda) [reply data/scripts/actions/mounts/mounts.lua:10]: My ref, is old =\ 
https://www.youtube.com/watch?v=Deupt7HWBhk

**Pull request closed without merging**